extensions/kadena/src/extension.ts

```diff
@@ -0,0 +1,147 @@
+import { ITatumSdkContainer, Network, TatumConfig, TatumSdkExtension } from '@tatumio/tatum'
+import {
+  ChainId,
+  createClient,
+  IUnsignedCommand,
+  Pact,
+  PactReference,
+  readKeyset
+} from "@kadena/client";
+
+import { KadenaGuard, KadenaNetworkId } from "./types";
+import {
+  COIN_GAS_CAPABILITY,
+  CREATE_TOKEN_ID_POLICIES,
+  NFT_PRECISION,
+  MARMALADE_CREATE_TOKEN_CAPABILITY,
+  MARMALADE_MINT_CAPABILITY,
+  NFT_AMOUNT,
+  MARMALADE_BURN_CAPABILITY,
+  NFT_BURN_AMOUNT,
+  MARMALADE_BURN_POLICY
+} from "./consts";
+import { KadenaLoadBalancerRpc } from "@tatumio/tatum/dist/src/service/rpc/other/KadenaLoadBalancerRpc";
+
+export class KadenaUtils extends TatumSdkExtension {
+  private readonly kadenaRpc: KadenaLoadBalancerRpc
+  private readonly sdkConfig: TatumConfig
+
+  constructor(tatumSdkContainer: ITatumSdkContainer) {
+    super(tatumSdkContainer)
+    this.sdkConfig = this.tatumSdkContainer.getConfig()
+    this.kadenaRpc = this.tatumSdkContainer.get(KadenaLoadBalancerRpc)
+  }
+
+  public async mintBasicNFT(uri: string | PactReference, publicKey: string, chainId: ChainId) : Promise<IUnsignedCommand> {
+
+    const guard : KadenaGuard = this.getGuardFromKey(publicKey)
+    const {networkId, client} = this.getTatumKadenaClient(chainId);
+
+    const createTokenIdCommand = Pact.builder
```

Comment:
Since a big portion of this will always be the same for us (`setMetadata`, `createTransaction`, etc.), maybe it would make sense to have it as a separate `util` function. Then it could also be used more easily even outside of these NFT related use cases.

Reply:
I'd rather have it all in one place till the chain implementation and ours hardens a bit more

Reply:
I don't mind having it in one place, but why not extract the parts that are always the same into a separate `util` function (even if it's `private` for now)?